src/main/java/it/gov/pagopa/payhub/activities/service/CsvService.java

```diff
@@ -70,6 +70,55 @@ public void createCsv(Path csvFilePath, List<String[]> header, List<String[]> da
         log.info("CSV file created successfully: {}", csvFilePath);
     }
 
+    /**
+     * Creates a CSV file from the provided iterator of beans.
+     *
+     * <p>This method ensures that the file is properly closed after processing
+     * by using a try-with-resources statement.</p>
+     *
+     * <p>This method uses {@code StatefulBeanToCsv.write()}
+     * to write each bean individually, reducing memory consumption at the cost of lower performance.</p>
+     *
+     * @param <C> the generic type of the beans to be written to the CSV
+     * @param csvFilePath the path to the CSV file to write
+     * @param typeClass the class type of the beans to be written to the CSV
+     * @param csvRowsSupplier a supplier of beans to be written to the CSV
+     * @param csvProfile the profile to be used for writing the CSV
+     * @throws IOException if an error occurs while writing the file
+     */
+    public <C> void createCsvFromBean(Path csvFilePath, Class<C> typeClass, Supplier<List<C>> csvRowsSupplier, String csvProfile) throws IOException {
+        if (csvFilePath == null || csvRowsSupplier == null) {
+            throw new IllegalArgumentException("Arguments cannot be null");
+        }
+
+        File file = csvFilePath.toFile();
+        File parentDir = file.getParentFile();
+        if (!parentDir.exists() && !parentDir.mkdirs()) {
+            throw new IOException("Unable to create directory: " + parentDir.getAbsolutePath());
+        }
+
+        try (Writer writer = Files.newBufferedWriter(csvFilePath)) {
+            HeaderColumnNameMappingStrategy<C> mappingStrategy = new HeaderColumnNameMappingStrategy<>();
+            mappingStrategy.setType(typeClass);
+            mappingStrategy.setProfile(csvProfile);
+
+            StatefulBeanToCsv<C> beanToCsv = new StatefulBeanToCsvBuilder<C>(writer)
+                    .withProfile(csvProfile)
+                    .withSeparator(separator)
+                    .withQuotechar(quoteChar)
+                    .withMappingStrategy(mappingStrategy)
+                    .withThrowExceptions(true)
+                    .build();
+
+            if (csvRowsSupplier.get() != Collections.emptyList()){
+                beanToCsv.write(csvRowsSupplier.get());
+            }
```

Comment:
you should call the supplier until it returns data (ps: document on the javadoc that the supplier is called more than once until return data)
(your service is expecting to have the supplier called many times changing the page at each invocation: thus you cannot call the .get as many times as you are doing here)